azure-quantum/azure/quantum/job/base_job.py

```diff
@@ -0,0 +1,240 @@
+        to blob storage
+
+        :param name: Job name
+        :type name: str
+        :param workspace: Azure Quantum workspace to submit the blob to
+        :type workspace: "Workspace"
+        :param job_id: Pre-generated job ID
+        :type job_id: str
+        :param target: Azure Quantum target
+        :type target: str
+        :param input_data_uri: Input data URI
+        :type input_data_uri: str
+        :param container_uri: Container URI
+        :type container_uri: str
+        :param input_data_format: Input data format, defaults to None
+        :type input_data_format: str, optional
+        :param output_data_format: Output data format, defaults to None
+        :type output_data_format: str, optional
+        :param provider_id: Provider ID, defaults to None
+        :type provider_id: str, optional
+        :param input_params: Input parameters, defaults to None
+        :type input_params: Dict[str, Any], optional
+        :return: Job instsance
+        :rtype: Job
+        """
+        if input_params is None:
+            input_params = {"params": {"timeout": DEFAULT_TIMEOUT}}
+        if input_data_format is None:
+            input_data_format = cls.input_data_format
+        if output_data_format is None:
+            output_data_format = cls.output_data_format
+        if provider_id is None:
+            provider_id = cls.provider_id
+
+        details = JobDetails(
+            id=job_id,
+            name=name,
+            container_uri=container_uri,
+            input_data_format=input_data_format,
+            output_data_format=output_data_format,
+            input_data_uri=input_data_uri,
+            provider_id=provider_id,
+            target=target,
+            input_params=input_params
+        )
+        return cls(workspace, details)
+
+    @staticmethod
+    def create_container(
+        workspace: "Workspace",
+        container_name: str
+    ):
+        # Create container URI and get container client
+        if workspace.storage is None:
+            # Get linked storage account from the service, create
+            # a new container if it does not yet exist
+            container_uri = workspace._get_linked_storage_sas_uri(
+                container_name
+            )
+            container_client = ContainerClient.from_container_url(
+                container_uri
+            )
+            create_container_using_client(container_client)
+        else:
+            # Use the storage acount specified to generate container URI,
+            # create a new container if it does not yet exist
+            container_uri = get_container_uri(
+                workspace.storage, container_name
+            )
+        return container_uri
+
+    @staticmethod
+    def upload_input_data(
+        container_uri: str,
+        input_data: bytes,
+        input_data_name = "inputData",
+        content_type = "application/json",
+        encoding = "",
```

Comment:
I'm not sure this method can assume anything about the content_type, so I wouldn't make this optional.